commons-profiler-api/src/main/java/ru/fix/commons/profiler/impl/ProfilerReporterImpl.java

```diff
@@ -54,25 +55,27 @@ public ProfilerReport buildReportAndReset() {
                 .stream()
                 .sorted()
                 .map(name -> buildReportAndReset(name, spentTime))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
                 .collect(Collectors.toList()));
         return report;
     }
 
 
-    private ProfilerCallReport buildReportAndReset(String name, long elapsed) {
+    private Optional<ProfilerCallReport> buildReportAndReset(String name, long elapsed) {
         SharedCounters counters = sharedCounters.get(name);
 
         long callsCount = counters.getCallsCount().sumThenReset();
         long sumStartStopLatency = counters.getSumStartStopLatency().sumThenReset();
 
         if (callsCount == 0) {
             cleanCounters(counters);
-            return new ProfilerCallReport(name);
+            return Optional.empty();
```

Comment:
`ProfilerReporterImpl#sharedCounters` will continuse to hold redundant records.  
It would be better to leave buildReportAndReset as is - it wil continue to return ProfilerCallReport  
but instead we can:  
1. remove empty calls from shared counters in `ProfilerReporterImpl#buildReportAndReset()`
2. after that in `ProfilerReporterImpl#buildReportAndReset()` call `ProfilerReporterImpl#buildReportAndReset(java.lang.String, long)` for keys that left in shared counters.

Be careful. it is not enough to simply remove key from concurrentMap,  we have to change `ProfiledCallImpl#stop(long)` method too so we will not lose metric. Seems like it is too hard to implement it without synchronization.  I will create bug for it and we will fix it next time.